core/src/main/java/com/linecorp/armeria/common/util/DefaultAsyncLoader.java

```diff
@@ -59,21 +62,28 @@ final class DefaultAsyncLoader<T> implements AsyncLoader<T> {
     private volatile CompletableFuture<T> loadingFuture = UnmodifiableFuture.completedFuture(null);
 
     DefaultAsyncLoader(Function<@Nullable T, CompletableFuture<T>> loader,
+                       @Nullable String name,
                        @Nullable Duration expireAfterLoad,
                        @Nullable Predicate<? super T> expireIf,
+                       @Nullable Duration refreshAfterLoad,
                        @Nullable Predicate<? super T> refreshIf,
                        @Nullable BiFunction<? super Throwable, ? super @Nullable T,
                                ? extends @Nullable CompletableFuture<T>> exceptionHandler) {
-        requireNonNull(loader, "loader");
+        executor = CommonPools.blockingTaskExecutor();
         this.loader = loader;
+        if (name == null) {
+            name = "AsyncLoader-" + Integer.toHexString(System.identityHashCode(this));
+        }
+        this.name = name;
         expireAfterLoadNanos = expireAfterLoad != null ? expireAfterLoad.toNanos() : 0;
         this.expireIf = expireIf;
+        refreshAfterLoadNanos = refreshAfterLoad != null ? refreshAfterLoad.toNanos() : 0;
         this.refreshIf = refreshIf;
         this.exceptionHandler = exceptionHandler;
     }
 
     @Override
-    public CompletableFuture<T> load() {
+    public CompletableFuture<T> load(boolean force) {
         final CompletableFuture<T> loadingFuture = this.loadingFuture;
         if (!loadingFuture.isDone()) {
             // A load is in progress.
```

Comment:
My initial expectation was that if `force = true`, a new load will be triggered. Depending on how this API is used, this may be important since
1) Currently, the actual asynchronous load is completed, but the future isn't completed.
2) Concurrently, a state modification is done and `load(true)` is called
3) Although `load(true)` is called, the load before state modification at step 2) is done

I understand the behavior can be different depending on the use case, but wanted to mention this